src/ProductService.php

```diff
@@ -627,6 +643,10 @@ public function buildProductsWithFallback(AmazonProductField $product_field) {
       '#title' => $products_container['title'],
       '#products' => $product_build,
     ];
+    
+    // Call alter hook so users can alter the data.
+    $this->moduleHandler->invokeAll('amazon_product_widget_alter_product_data', [&$products_container, $node, $product_field]);
```

Comment:
this is redundant, as will happen before in getProductsWithFallback anyways

Reply:
also above,buildProductsWithFallback is not getting the node and also not passing it through to getProductsWithFallback